src/index.ts

```diff
@@ -562,8 +575,22 @@ class LayercodeClient implements ILayercodeClient {
       // Reset turn tracking for clean start
       this._resetTurnTracking();
       this._stopAmplitudeMonitoring();
+
+      // Safari-specific race condition fix:
+      // _setupDeviceChangeListener() calls setInputDevice() which is async.
+      // We wait for the recorder to be initialized before proceeding with WebSocket connection.
+      // Without this, we'd send client.ready before the recorder is actually ready,
+      // causing the agent to start speaking before we can capture user audio.
+      this.recorderStartedPromise = new Promise<void>((resolve) => {
+        this.recorderStartedResolve = resolve;
+      });
+
+      this.requestPermissions();
       this._setupDeviceChangeListener();
 
+      // Wait max 3 seconds for device setup to prevent indefinite hanging
+      await Promise.race([this.recorderStartedPromise, new Promise<void>((resolve) => setTimeout(resolve, 3000))]);
```

Comment:
this solves a separate issue where we're not awaiting device readiness before we send client.ready to the pipeline